File: events/get_profile_events.py
# this sample Python code retrieves all events for a specific profile. you can then use these events to power an activity feed.
import os
from klaviyo_api import KlaviyoAPI
from datetime import datetime

# load your private API key from environment variables. 
# ensure you have set the environment variable "private_api_key" with Klaviyo private API key that has read access to events and profiles. 
private_api_key = os.environ.get("private_api_key")

# instantiate the KlaviyoAPI client with your private API key.
klaviyo = KlaviyoAPI(private_api_key)

# set customer_id to the profile ID of the profile whose events you are looking for.
customer_id = '01G9ATZKKMSZMDB76EN6WVMVGM'

klaviyo.Events.get_events(filter=f"equals(profile_id,'{customer_id}')")

# if you do not know the profile ID, first use Get Profiles to retrieve profile ID, then use it in the Get Events request.
# set customer_email to the email address of the profile whose events you are looking for.
customer_email = 'michaela.klaviyo@gmail.com'

# use the Get Profiles endpoint to retrieve the profile ID using the email address.
profile = klaviyo.Profiles.get_profiles(filter=f"equals(email,'{customer_email}')")
profile_id = profile.data[0].id

klaviyo.Events.get_events(filter=f"equals(profile_id,'{profile_id}')")

File: events/get_profile_events.js
// this sample JavaScript code retrieves all events for a specific profile. you can then use these events to power an activity feed.

import { ApiKeySession, EventsApi, ProfilesApi } from 'klaviyo-api'

// load private API key as an environment variable. Note that this API key needs to have read access to events and profiles.
// if using a .env file, ensure to load it using dotenv or similar package. 
const apiKey = process.env.KLAVIYO_API_KEY
if (!apiKey) {
  throw new Error('Please set the KLAVIYO_API_KEY environment variable.')
}

const session = new ApiKeySession(apiKey)
const eventsApi = new EventsApi(session)

// set customerId to the ID of the profile you want to retrieve events for
const customerId = '01G9ATZKKMSZMDB76EN6WVMVGM'

const filter = `equals(profile_id,'${customerId}')`
const profileEvents = await eventsApi.getEvents({ filter })
console.log('Profile Events:', profileEvents)

// if you do not know the customer's profile ID, first use Get Profiles to retrieve profile ID, then use it in the Get Events request.

// instantiate profilesAPI
const profilesApi = new ProfilesApi(session)

// set customerEmail to the email address of the profile whose events you are looking for.
const customerEmail = 'michaela.klaviyo@gmail.com'
const profileFilter = `equals(email,'${customerEmail}')`

// use Get Profiles to retrieve the profile ID based on the email address
const profile = await profilesApi.getProfiles({ filter: profileFilter })
const profileId = profile.body.data[0].id

// use Get Events to retrieve all events for the profile ID
const eventsFilter = `equals(profile_id,'${profileId}')`
const profileEventsByEmail = await eventsApi.getEvents({ filter: eventsFilter })
console.log('Profile Events by Email:', profileEventsByEmail)